legacy/application/models/ShowInstance.php

```diff
@@ -844,4 +844,30 @@ public function isRepeating()
     {
         return $this->getShow()->isRepeating();
     }
+
+    public function trimOverbooked()
+    {
+        // iterate through all of the showinstance items and delete those with start time greater than endtime of the show
+        $showEnd = new DateTime($this->getShowInstanceEnd(), new DateTimeZone('UTC'));
+        $removeItems = [];
+        foreach ($this->getShowListContent('UTC') as $item) {
+            $itemStart = new DateTime($item['starts'], new DateTimeZone('UTC'));
+            if ($itemStart->getTimestamp() > $showEnd->getTimestamp()) {
+                $removeItems[] = $item['id'];
+            }
+        }
+
+        // now remove the items
+        $sql = <<<'SQL'
+        DELETE FROM cc_schedule
+         WHERE id in (:ids)
+           AND instance_id = :instance_id
+           AND playout_status >= 0
+        SQL;
```

Comment:
We could also make this in a single sql only call, without the loops:
```
delete
from
	cc_schedule
where
	id in (
	select
		s.id
	from
		cc_schedule s
	left join cc_show_instances si on
		s.instance_id = si.id
	where
		si.id = :instance_id
		and si.ends < s.starts
		and s.playout_status = 0 -- Double check si.ends < s.starts
);
```

Reply:
That call would work too ... for my mental sanity i try to avoid those. The loop does not do database lookups and the code is identical to what the UI button does. IIRC the crossfade is already calculated from the cuepoints of the mp3 files when the show is populated, so it will work either way.